src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -188,6 +188,54 @@ public function getBoolean($key, $default = false)
     }
 
     /**
+     * Returns the parameter value converted to a DateTime object.
+     *
+     * @param string             $key      The parameter key
+     * @param string             $format   The expected date format
+     * @param \DateTime|null     $default  The default value if the parameter key does not exist
+     * @param \DateTimeZone|null $timeZone
+     *
+     * @return \DateTime|null
+     */
+    public function getDate($key, $format = 'Y-m-d', $default = null, $timeZone = null)
+    {
+        if (!$this->has($key)) {
+            return $default;
```

Comment:
This gives no guaranties whatsoever about the type of the return value ($default can be anything).
Should default be a plain string that is turned to a date instead?